File: .tmp_sc4013/java_06/2225_find_players_with_zero_or_one_losses.java
// LeetCode 2225 - Find Players With Zero or One Losses
// https://leetcode.com/problems/find-players-with-zero-or-one-losses/

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Map<Integer, Integer> lose = new HashMap<>();
        Set<Integer> seen = new HashSet<>();
        for (int[] m : matches) {
            seen.add(m[0]);
            seen.add(m[1]);
            lose.put(m[1], lose.getOrDefault(m[1], 0) + 1);
        }
        List<Integer> zero = new ArrayList<>();
        List<Integer> one = new ArrayList<>();
        for (int p : seen) {
            int L = lose.getOrDefault(p, 0);
            if (L == 0) zero.add(p);
            else if (L == 1) one.add(p);
        }
        Collections.sort(zero);
        Collections.sort(one);
        List<List<Integer>> ans = new ArrayList<>();
        ans.add(zero);
        ans.add(one);
        return ans;
    }
}

File: .tmp_py4013/batch_02_sources/2225_find_players_with_zero_or_one_losses.js
// LeetCode 2225 - Find Players With Zero or One Losses
// https://leetcode.com/problems/find-players-with-zero-or-one-losses/

/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function(matches) {
    const lose = new Map();
    const seen = new Set();
    for (const m of matches) {
        seen.add(m[0]);
        seen.add(m[1]);
        lose.set(m[1], (lose.get(m[1]) || 0) + 1);
    }
    const zero = [], one = [];
    for (const p of seen) {
        const L = lose.get(p) || 0;
        if (L === 0) zero.push(p);
        else if (L === 1) one.push(p);
    }
    zero.sort((a, b) => a - b);
    one.sort((a, b) => a - b);
    return [zero, one];
};
